File: cpp/41a_1579A_CasimirsStringSolitaire.cpp
#include<bits/stdc++.h>

int main(){
    int t;
    std::string s;
    std::vector<std::string> ans;
    std::cin>> t;
    for(int i = 0; i < t; i++){
        std::cin>>s;
        std::unordered_map<char, int> hmap;
        for(auto j : s){
            hmap[j]++;
        }
        if(hmap['A'] + hmap['C'] == hmap['B']){
            ans.push_back("YES");
        }
        else{
            ans.push_back("NO");
        }
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/41a_1579A_CasimirsStringSolitaire.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        s = input().strip()
        hmap = {"A" : 0, "B" : 0, "C" : 0}
        for i in s:
            hmap[i] += 1
        if hmap["A"] + hmap["C"] == hmap["B"]:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()